chandra_aca/star_probs.py

```diff
@@ -227,24 +247,167 @@ def acq_success_prob(date=None, t_ccd=-10.0, mag=10.0, color=0.6, spoiler=False,
     p_0p7color = .4294  # probability multiplier for a B-V = 0.700 star (REF?)
     p_spoiler = .9241  # probability multiplier for a search-spoiled star (REF?)
 
-    # If the star is brighter than 8.5 or has a calculated probability
-    # higher than the max_star_prob, clip it at that value
     probs[np.isclose(colors, 0.7, atol=1e-6, rtol=0)] *= p_0p7color
     probs[spoilers] *= p_spoiler
 
-    probs = probs.clip(MIN_ACQ_PROB, MAX_ACQ_PROB)
+    if model in ('sota', 'spline'):
+        # Clip probabilities for older models.  Newer models (grid-* included)
+        # should do all this internally.
+        probs = probs.clip(MIN_ACQ_PROB, MAX_ACQ_PROB)
 
     # Return probabilities.  The [()] getitem at the end will flatten a
     # scalar array down to a pure scalar.
     return probs[0] if is_scalar else probs
 
 
+def clip_and_warn(name, val, val_lo, val_hi, model):
+    """
+    Clip ``val`` to be in the range ``val_lo`` to ``val_hi`` and issue a
+    warning if clipping occurs.  The ``name`` and ``model`` are just used in
+    the warning.
+
+    :param name: Value name
+    :param val: Value
+    :param val_lo: Minimum
+    :param val_hi: Maximum
+    :param model: Model name
+
+    :returns: Clipped value
+
+    """
+    val = np.asarray(val)
+    if np.any((val > val_hi) | (val < val_lo)):
+        warnings.warn('\nModel {} computed between {} <= {} <= {}, '
+                      'clipping input {}(s) outside that range.'
+                      .format(model, name, val_lo, val_hi, name))
+        val = np.clip(val, val_lo, val_hi)
+
+    return val
+
+
+def grid_model_acq_prob(mag=10.0, t_ccd=-12.0, color=0.6, halfwidth=120, probit=False,
+                        model=None):
+    """Calculate a grid model probability of acquisition success for a star with
+    specified mag, t_ccd, color, and search box halfwidth.
+
+    This does a 3-d linear interpolation on mag, t_ccd, and halfwidth using a
+    pre-computed gridded model that is stored in a FITS file.
+
+    :param mag: ACA magnitude (float or np.ndarray)
+    :param t_ccd: CCD temperature (degC, float or ndarray)
+    :param color: B-V color to check for B-V=1.5 => red star (float or np.ndarray)
+    :param halfwidth: search box halfwidth (arcsec, default=120, float or ndarray)
+    :param probit: if True then return Probit(p_success). Default=False
+    :param model: Model name, e.g. 'grid-floor-2018-11'
+
+    :returns: Acquisition success probability(s)
+
+    """
+    # Info about model is cached in GRID_FUNCS
+    if model not in GRID_FUNCS:
+        from astropy.io import fits
+        from scipy.interpolate import RegularGridInterpolator
+
+        # Read the model file and put into local vars
+        filename = os.path.join(STAR_PROBS_DATA_DIR, model) + '.fits.gz'
+        if not os.path.exists(filename):
+            raise IOError('model file {} does not exist'.format(filename))
+
+        hdus = fits.open(filename)
```

Comment:
Also, why fits?  Fine with it, I just thought you hate fits files.

Reply:
FITS is a fine format for storing images with simple metadata, and it is the easiest for me from Python with astropy.